superset/models/core.py

```diff
@@ -438,12 +442,29 @@ def _log_query(sql: str) -> None:
         with closing(engine.raw_connection()) as conn:
             cursor = conn.cursor()
             for sql_ in sqls[:-1]:
+                if mutate_after_split:
+                    sql_ = sql_query_mutator(
+                        sql_,
+                        user_name=username,
+                        security_manager=security_manager,
+                        database=None
+                    )
                 _log_query(sql_)
                 self.db_engine_spec.execute(cursor, sql_)
                 cursor.fetchall()
-
-            _log_query(sqls[-1])
-            self.db_engine_spec.execute(cursor, sqls[-1])
+            
+            if mutate_after_split:
+                last_sql = sql_query_mutator(
+                    sqls[-1],
+                    user_name=username,
+                    security_manager=security_manager,
+                    database=None
+                )
+                _log_query(last_sql)
+                self.db_engine_spec.execute(cursor, last_sql)
+            else:
+                _log_query(sqls[-1])
+                self.db_engine_spec.execute(cursor, sqls[-1])
```

Comment:
It appears `sqls[-1]` is needed in both branches of the if statement. This should simplify it somewhat
```suggestion
            last_sql = sqls[-1]
            if mutate_after_split:
                last_sql = sql_query_mutator(
                    last_sql,
                    user_name=username,
                    security_manager=security_manager,
                    database=None
                )
                _log_query(last_sql)
                self.db_engine_spec.execute(cursor, last_sql)
            else:
                _log_query(last_sql)
                self.db_engine_spec.execute(cursor, last_sql)
```